File: json-schema-validation/src/test/java/io/micronaut/jsonschema/validation/Llama.java
package io.micronaut.jsonschema.validation;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import io.micronaut.jsonschema.JsonSchema;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.PositiveOrZero;

/**
 * A llama. <4>
 *
 * @param name The name
 * @param age The age
 */
@JsonSchema // <1>
public record Llama(
    @NotBlank // <3>
    @JsonInclude(Include.NON_NULL)
    String name,
    @PositiveOrZero // <3>
    int age
) {
}

File: test-suite-python/src/test/python/micronaut/jsonschema/test/Llama.py
from dataclasses import dataclass
from typing import Annotated

from com.fasterxml.jackson.annotation import JsonInclude
from jakarta.validation.constraints import NotBlank, PositiveOrZero
from micronaut.jsonschema import JsonSchema
from micronaut.serde.annotation import Serdeable


@JsonSchema  # <1>
@Serdeable  # <2>
@dataclass
class Llama:
    """A llama."""  # <4>

    name: Annotated[str, NotBlank, JsonInclude(JsonInclude.Include.NON_NULL)]  # <3>
    """The name"""

    age: Annotated[int | None, PositiveOrZero] = None  # <3>
    """The age"""
